### PR TITLE
wait for ArethusaLoaded event

### PR DESCRIPTION
**Pull request by balmas**
I decided not to add a new api call to the treebank template for this, just to have it send a SERVICE_UNINITIALIZED error to any request received before the ArethusaLoaded event was received.

I think this means the interval for setting the subdoc can be gotten rid of too. Let me know if these changes don't look right to you though.

### CODE REVIEW

File: src/components/Publication/Publication.js [hunk 1]
```diff
@@ -76,31 +76,28 @@ class Publication extends Component {
...
   }
 
   componentWillUnmount() {
-    // eslint-disable-next-line no-undef
-    window.clearInterval(this.interval);
+    document.body.removeEventListener('ArethusaLoaded');
   }
 
+  setArethusaLoaded() {
+    document.body.removeEventListener('ArethusaLoaded');
```
Comment (zfletch) [inline src/components/Publication/Publication.js]: I'm getting the error `Failed to execute 'removeEventListener' on 'EventTarget': 2 arguments required, but only 1 present` in my console from this line.

File: src/components/Publication/Publication.js [hunk 2]
```diff
@@ -76,31 +76,28 @@ class Publication extends Component {
...
+    this.setSubdoc();
```
Comment (zfletch) [inline src/components/Publication/Publication.js]: One thing we'll have to do now that we're removing the timer is to make sure that `this.setSubdoc()` runs when a user clicks `Next` or `Back`. We should be able to use [componentDidUpdate](https://reactjs.org/docs/react-component.html#componentdidupdate). We'd still need to guard against the case where Arethusa is not loaded, but we can use something like this code:

```javascript
componentDidUpdate(_prevProps, prevState) {
  const newSubDoc = this.arethusa.getSubdoc();
  const { subDoc } = prevState;

  if (subDoc !== newSubDoc) {
    this.setState({ subDoc: newSubDoc });
  }
}
```

File: public/arethusa/arethusa.packages.min.js [hunk 3]
```diff
@@ -388,11 +387,11 @@ var Z=s();"function"==typeof define&&"object"==typeof define.amd&&define.amd?(H.
...
 //# sourceMappingURL=toaster.min.map
-/*! angular-highlightjs
-version: 0.3.3
-build date: 2015-01-11
-author: Robin Fan
-https://github.com/pc035860/angular-highlightjs.git */
+/*! angular-highlightjs
+version: 0.3.3
```
Comment (zfletch) [inline public/arethusa/arethusa.packages.min.js]: This file is the one that has the modified Angular code necessary to work within the React app. It needs to be reverted to the version in `master`.

**Comment by balmas** [comment]
> This is great! I think we can get it working with just some small changes. I'll also have time next week to help out and make some of the changes.

Thanks for your review! I have made all of the requested changes except for the ones for making sure setSubdoc is called upon Next/Prev. I think I will let you do those if you don't mind :-)

**Comment by balmas** [comment]
FYI, I've updated the PR to include the latest Arethusa change for alpheios-project/arethusa#796

**Comment by zfletch** [comment]
These changes were incorporated in https://github.com/perseids-publications/treebank-template/pull/22

**Pull request closed without merging**